### PR TITLE
Managed acceptance for existent nodes floc 3387

### PR DESCRIPTION
**Pull request by mpgomez**
Review comments implemented and code cleaned up

### CODE REVIEW

**Comment by mpgomez** [comment]
Green build still pending, but there are quite a few intermittent failures right now, so I am sending the code for review anyway - I think I've already fixed the tests that were failing because of my changes, but I won't merge with master anyway until I get a green build

File: admin/acceptance.py [hunk 1]
```diff
@@ -298,8 +298,9 @@ def configure(ignored):
                 self.dataset_backend,
                 self.dataset_backend_configuration,
                 _save_backend_configuration(self.dataset_backend,
-                                            self.dataset_backend_configuration)
-            )
+                                            self.dataset_backend_configuration),
+                provider="managed"
+                )
```
Comment (jml) [inline admin/acceptance.py]: Nit: closing paren should be dedented by 4, as in original version.

File: admin/acceptance.py [hunk 2]
```diff
@@ -314,6 +315,8 @@ def _provider_for_cluster_id(dataset_backend):
     """
     Get the ``Providers`` value that probably corresponds to a value from
     ``DatasetBackend``.
+    Note that this function will ignore the case of a managed provider,
+    as this information cannot be known just knowing the backend.
```
Comment (jml) [inline admin/acceptance.py]: Please add "by" after "just".

File: flocker/provision/_install.py [hunk 3]
```diff
@@ -825,12 +847,25 @@ def task_enable_flocker_control(distribution):
...
 
+def validate_start_action(action):
+    """
+    validates if the action given is a valid one  - currently only
+    start and restart are supported
+    """
+    valid_actions = ["start", "restart"]
+    for valid_action in valid_actions:
+        if action.lower() == valid_action:
+            return
```
Comment (jml) [inline flocker/provision/_install.py]: You can condense this just to:

```python
if action.lower() in valid_actions:
    return
```

Once that's done, then it probably makes more sense to invert the conditional and drop the explicit return, e.g.:

```python
if action.lower() not in valid_actions:
    raise UnknownAction(action)
```

File: flocker/provision/_install.py [hunk 4]
```diff
@@ -825,12 +847,25 @@ def task_enable_flocker_control(distribution):
...
+            run_from_args(['service', 'flocker-control', action.lower()]),
         ])
+
     else:
         raise DistributionNotSupported(distribution=distribution)
 
 
+def validate_start_action(action):
+    """
+    validates if the action given is a valid one  - currently only
```
Comment (jml) [inline flocker/provision/_install.py]: Nit: Capital 'V' for "Validates".

File: flocker/provision/test/test_install.py [hunk 5]
```diff
@@ -95,6 +95,25 @@ def test_centos_sequence(self):
...
+            run(command="systemctl enable flocker-container-agent"),
+            run(command="systemctl start flocker-container-agent"),
+        ])
+        self.assertEqual(commands, expected_sequence)
+
+    def test_centos_sequence_managed(self):
+        """
+        ``task_enable_flocker_agent`` for the 'centos-7' distribution
+        returns a sequence of 'service restart' commands for each agent
+        when the action passed down is "restart" (used for managed provider)..
```
Comment (jml) [inline flocker/provision/test/test_install.py]: Nit: Two full stops at the end of sentence, should be one.

File: flocker/provision/test/test_install.py [hunk 6]
```diff
@@ -116,6 +135,45 @@ def test_ubuntu_sequence(self):
...
+        expected_sequence = sequence([
+            run(command="service flocker-dataset-agent restart"),
+            run(command="service flocker-container-agent restart"),
+        ])
+        self.assertEqual(commands, expected_sequence)
+
+    def test_sequence_invalid_action(self):
+        """
+        ``task_enable_flocker_agent`` for a valid distribution
+        but a non valid action raises a ``UnknownAction``.
```
Comment (jml) [inline flocker/provision/test/test_install.py]: s/a non valid/an invalid/

**Comment by jml** [comment]
Thanks Pilar! Pretty much all of my comments are minor nitpicks, plus one small refactoring. Please address them & merge when the tests are passing.

**Comment by mpgomez** [comment]
Green build! Checked in!

**Pull request merged**